### ISSUE TITLE
An internal error occurred while type checking file: RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by lrtfm** [microsoft/pylance-release#5354]
## Environment data

-   Language Server version: 2023.12.1 (pyright a78e2b10)
-   OS and version: darwin arm64
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.11.6
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: off

## Code Snippet

I am using the package [firedrake](https://github.com/firedrakeproject/firedrake), which installed itself in an isolated environment by `venv`. It can be installed by following the instructions here: https://www.firedrakeproject.org/download.html

```python
from firedrake import *

mesh = UnitSquareMesh(2, 2)
V = FunctionSpace(mesh, "CG", 1)
x, y = SpatialCoordinate(mesh)

f = Function(V)
interpolate(x, f)
```

## Repro Steps

1. Install firedrake OR use the docker image `docker pull firedrakeproject/firedrake:latest`
2. Open the py file in vscode 
3. Select the interpreter from the firedrake environment

## Actual behavior
<img width="844" alt="截屏2024-01-16 17 35 58" src="https://github.com/microsoft/pylance-release/assets/1192812/187c82c0-861f-4d67-9f55-375bfe5a256c">

## Logs

Part of the log is shown below. For the whole log: see [pylance.log](https://github.com/microsoft/pylance-release/files/13948112/pylance.log)

```
2024-01-16 17:22:08.691 [info] (Client) Pylance async client (2023.12.1) started with python extension (2023.22.1)
2024-01-16 17:22:09.078 [info] [Info  - 17:22:09] (19410) Pylance language server 2023.12.1 (pyright a78e2b10) starting
2024-01-16 17:22:09.079 [info] [Info  - 17:22:09] (19410) Server root directory: /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist
2024-01-16 17:22:09.080 [info] [Info  - 17:22:09] (19410) Starting service instance "debug"
2024-01-16 17:22:09.116 [info] (19410) No configuration file found.
2024-01-16 17:22:09.116 [info] (19410) No pyproject.toml file found.
2024-01-16 17:22:09.116 [info] [Info  - 17:22:09] (19410) Setting pythonPath for service "debug": "/Users/zzyang/opt/firedrake/firedrake-real-int32-debug/bin/python"
2024-01-16 17:22:09.116 [info] [Info  - 17:22:09] (19410) Setting environmentName for service "debug": "3.11.6 (firedrake-real-int32-debug venv)"
2024-01-16 17:22:09.148 [info] [Info  - 17:22:09] (19410) Assuming Python version 3.11
2024-01-16 17:22:09.148 [info] (19410) Assuming Python platform Darwin
2024-01-16 17:22:09.250 [info] [Info  - 17:22:09] (19410) Search paths for /Users/zzyang/workspace/test/debug
2024-01-16 17:22:09.250 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/typeshed-fallback/stdlib
2024-01-16 17:22:09.250 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/workspace/test/debug
2024-01-16 17:22:09.251 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/workspace/test/debug/typings
2024-01-16 17:22:09.251 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/typeshed-fallback/stubs/...
2024-01-16 17:22:09.251 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/bundled/stubs
2024-01-16 17:22:09.252 [info] [Info  - 17:22:09] (19410)   /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11
2024-01-16 17:22:09.252 [info] [Info  - 17:22:09] (19410)   /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/lib-dynload
2024-01-16 17:22:09.253 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/lib/python3.11/site-packages
2024-01-16 17:22:09.253 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/pytest-mpi
2024-01-16 17:22:09.253 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/fdutils
2024-01-16 17:22:09.253 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/loopy
2024-01-16 17:22:09.253 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/fiat
2024-01-16 17:22:09.253 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/FInAT
2024-01-16 17:22:09.253 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/tsfc
2024-01-16 17:22:09.254 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/pyadjoint
2024-01-16 17:22:09.254 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/PyOP2
2024-01-16 17:22:09.254 [info] [Info  - 17:22:09] (19410)   /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/firedrake
2024-01-16 17:22:09.254 [info] [Info  - 17:22:09] (19410) Adding fs watcher for library directories:
 /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11
/opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/lib-dynload
/Users/zzyang/opt/firedrake/firedrake-real-int32-debug/lib/python3.11/site-packages
/Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/pytest-mpi
/Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/fdutils
/Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/loopy
/Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/fiat
/Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/FInAT
/Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/tsfc
/Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/pyadjoint
/Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/PyOP2
/Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/firedrake
2024-01-16 17:22:09.254 [info] [Info  - 17:22:09] (19410) Adding fs watcher for directories:
 /Users/zzyang/workspace/test/debug
...
2024-01-16 17:22:11.398 [info] [Info  - 17:22:11] (19410)   Attempting to resolve using root path '/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/typeshed-fallback/stdlib'
2024-01-16 17:22:11.398 [info] [Info  - 17:22:11] (19410)   Typeshed path not found
2024-01-16 17:22:11.398 [info] [Info  - 17:22:11] (19410)   Looking for typeshed third-party path
2024-01-16 17:22:11.398 [info] [Info  - 17:22:11] (19410)   Looking for typeshed stubs path
2024-01-16 17:22:11.399 [info] [Info  - 17:22:11] (19410)   Typeshed path not found
2024-01-16 17:22:11.399 [info] (19410) [BG(1)]     binding: /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/firedrake/firedrake/adjoint_utils/blocks/function.py (1ms)
2024-01-16 17:22:11.399 [info] (19410) [BG(1)]     parsing: /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/PyOP2/pyop2/exceptions.py [fs read 1ms] (1ms)
2024-01-16 17:22:11.399 [info] (19410) [BG(1)]     binding: /Users/zzyang/opt/firedrake/firedrake-real-int32-debug/src/PyOP2/pyop2/exceptions.py (0ms)
2024-01-16 17:22:11.399 [info] [Error - 17:22:11] (19410) An internal error occurred while type checking file "/Users/zzyang/workspace/test/debug/debug_pylance.py": RangeError: Maximum call stack size exceeded
    at O (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:917246)
    at M (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:917190)
    at St (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:622123)
    at Ge (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591467)
    at Ke (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591985)
    at Ka (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:816827)
    at ri (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:843090)
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:842103
    at t.mapSignatures (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:928596)
    at ii (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:841871)
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:630085
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:630289
    at d (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:911129)
    at St (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:625579)
    at Ge (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591467)
    at Ke (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591985)
    at Ka (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:816827)
    at ri (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:843090)
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:842103
    at t.mapSignatures (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:928596)
    at ii (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:841871)
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:630085
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:630289
    at d (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:911129)
    at St (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:625579)
    at Ge (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591467)
    at Ke (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591985)
    at Ka (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:816827)
    at ri (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:843090)
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:842103
    at t.mapSignatures (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:928596)
    at ii (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:841871)
2024-01-16 17:22:11.399 [info] (19410) [BG(1)]   checking: /Users/zzyang/workspace/test/debug/debug_pylance.py (375ms)
2024-01-16 17:22:11.399 [info] (19410) [BG(1)] analyzing: /Users/zzyang/workspace/test/debug/debug_pylance.py (376ms)
2024-01-16 17:22:11.399 [info] [Error - 17:22:11] (19410) Error performing analysis: RangeError: Maximum call stack size exceeded
    at O (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:917246)
    at M (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:917190)
    at St (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:622123)
    at Ge (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591467)
    at Ke (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591985)
    at Ka (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:816827)
    at ri (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:843090)
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:842103
    at t.mapSignatures (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:928596)
    at ii (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:841871)
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:630085
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:630289
    at d (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:911129)
    at St (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:625579)
    at Ge (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591467)
    at Ke (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591985)
    at Ka (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:816827)
    at ri (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:843090)
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:842103
    at t.mapSignatures (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:928596)
    at ii (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:841871)
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:630085
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:630289
    at d (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:911129)
    at St (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:625579)
    at Ge (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591467)
    at Ke (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:591985)
    at Ka (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:816827)
    at ri (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:843090)
    at /Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:842103
    at t.mapSignatures (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:928596)
    at ii (/Users/zzyang/.vscode/extensions/ms-python.vscode-pylance-2023.12.1/dist/pyright.bundle.js:1:841871)
2024-01-16 17:22:11.400 [info] (19410) [BG(1)] indexing: /Users/zzyang/workspace/test/debug/debug_pylance.py [found 5] (1ms)
2024-01-16 17:22:11.401 [info] (19410) [BG(1)] getSemanticTokens range 0:0 - 7:17 at /Users/zzyang/workspace/test/debug/debug_pylance.py (101ms)
2024-01-16 17:22:11.401 [info] (19410) pytest configurations: {"message":"script","classes":["Test"],"files":["test_*.py","*_test.py"],"functions":["test"]}

```

**Comment by debonte** [comment]
I'm having trouble installing firedrake on my Linux box. Some sort of network issue on my side. But I suspect that this is a duplicate of https://github.com/microsoft/pyright/issues/6783. If so, the fix will be included in our next prerelease build.

**Comment by lrtfm** [comment]
Thank you for providing the information. I'll check if this issue is addressed in the upcoming prerelease. If it is resolved, I will proceed to close this.

**Comment by debonte** [comment]
@lrtfm, we shipped prerelease build 2023.12.104 this morning. Please check to see if your issue is fixed there.

**Comment by lrtfm** [comment]
@debonte, it has been fixed! Thanks!
